biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -1751,6 +1755,23 @@ void augmentImports(Packages imports, Packages exports) throws Exception {
 		}
 	}
 
+	void moveDynamicImports() {
+		Parameters dynamicImportPackages = getDynamicImportPackage();
+		Iterator<Entry<PackageRef,Attrs>> importsIterator = imports.entrySet().iterator();
+		while (importsIterator.hasNext()) {
+			Entry<PackageRef,Attrs> packageEntry = importsIterator.next();
+			PackageRef packageRef = packageEntry.getKey();
+			Attrs attrs = packageEntry.getValue();
+			String resolution = attrs.get(Constants.RESOLUTION_DIRECTIVE);
+			if (Constants.DYNAMIC_ATTRIBUTE.equals(resolution)) {
+				attrs.remove(Constants.RESOLUTION_DIRECTIVE);
+				dynamicImportPackages.add(packageRef.getFQN(), attrs);
+				importsIterator.remove();
```

Comment:
I am not sure this is exactly the right way to go to remove these from `imports`. There are other places where `imports` is checked and it should still contain all the imports. You should do the division into `Import-Package` and `DynamicImport-Package` in `calcManifest()` but `imports` should always retain all the imported packages so plugins which run after the analyzer can see all the imports.